### ISSUE TITLE
[Question] Play/resume first, next episode button

### ISSUE THREAD

**Issue by matke-84** [jurialmunkey/plugin.video.themoviedb.helper#795]
Hi Jurial. Is there a possibility to get a button in the info dialog window (but in tv show level, in episode level works) that would start or resume the first episode and then every next unwatched episode. This works with the local library but not with the tmdb helper and I have absolutely no idea how to do it or if it is even possible with tmdb helper. Currently, in order to get to the episode I want to watch, I have to go to more episodes, seasons and finally to the episode I want. This button would shorten path like I did with the local library. Thanks in advance.

**Comment by jurialmunkey** [comment]
Playing the first episode is easy if you have the tvshow tmdb_id (and if you don't you can use query= just as anywhere else).

`PlayMedia(plugin://plugin.video.themoviedb.helper/?info=play&amp;tmdb_type=tv&amp;tmdb_id=$INFO[ListItem.UniqueID(tmdb)]&amp;season=1&amp;episode=1)`

Playing a playlist depends on the plugin used to play. TMDbHelper cannot control this behaviour. 

TMDbHelper provides a `make_playlist` flag which can be added to player files in configure players but whether it works depends on the end plugin not hacking playback and using setResolvedURL properly as Kodi expects.

**Comment by matke-84** [comment]
OK. It's a pity that there is no way to overcome this. That would be a very good option. But this with the first episode is just as useful too. How could I restrict the button to only be visible when opened in the tmdb helper?

**Comment by matke-84** [comment]
@jurialmunkey
I guess it is not even possible to auto hide button when I have watched the first episode.

**Comment by jurialmunkey** [comment]
> OK. It's a pity that there is no way to overcome this. That would be a very good option.

It already is an option.

The way to overcome it is by plugin authors doing things the correct way via setResolvedURL and not hacking the player. There's no point TMDbH queuing a playlist if the end plugin hacks xbmc.Player() and overwrites the queue.

None of these 3rd party plugins go through code review, so they all use the same hacks rather than learning to do things the proper way, or submitting patches to Kodi core to add missing features. They just keep digging a hole of hacks on hacks.

> How could I restrict the button to only be visible when opened in the tmdb helper

You can check what plugin a listitem comes from by checking its path:

```xml
<visible>String.IsEqual(ListItem.Path,plugin://plugin.video.themoviedb.helper/)</visible>
```

>  I guess it is not even possible to auto hide button when I have watched the first episode.

You could check watchedepisodes count. It's not entirely accurate but generally I would expect that if you've watched at least one episode then you've watched the first episode (since typically you start at the beginning!).

```xml
<visible>Integer.IsEqual(ListItem.Property(watchedepisodes),0)</visible>
```

**Comment by jurialmunkey** [comment]
In addition to the above, don't forget that the UpNext season folder exists. You can easily add this as a container hidden offscreen and use it to supply the button action to get the Next episode to watch

Here's some code to get you started:

```xml
<control type="list" id="1234">
    <width>1</width>
    <height>1</height>
    <left>-1920</left>
    <content>plugin://plugin.video.themoviedb.helper/?info=trakt_upnext&amp;tmdb_id=$INFO[ListItem.UniqueID(tmdb)]&amp;tmdb_type=tv</content>
    <visible>String.IsEqual(ListItem.DBType,tvshow) | String.IsEqual(ListItem.DBType,season)</visible>
    <visible>!String.IsEmpty(ListItem.UniqueID(tmdb))</visible>
</control>

<control type="button">
    <visible>String.IsEqual(ListItem.DBType,tvshow) | String.IsEqual(ListItem.DBType,season)</visible>
    <visible>!Container(1234).IsUpdating + !Integer.IsEqual(Container(1234).NumItems,0)</visible>
    <visible>!String.IsEmpty(ListItem.UniqueID(tmdb))</visible>
    <onclick>PlayMedia($INFO[Container(1234).ListItemAbsolute(0).FolderPath])</onclick>
</control>
```

**Comment by matke-84** [comment]
@jurialmunkey 
Thanks a lot. I completely forgot about up next folder.
And just one more question. In the extended info dialog (not tmdbh extended info), I also set a button to start tmdb helper when watching a movie or tv show. What condition should I set in extendedinfo xml so that it can be seen only when I start a movie or tv show from tmdb helper. I tried a couple of variants with the plugin path as you gave me for the info dialog, but it didn't work.
